src/components/backgroundselector/BackgroundService.js

```diff
@@ -116,7 +124,7 @@ goog.require('ga_urlutils_service');
         } */
       };
 
-      function updateBgLayerStyleUrlParam(bgLayer) {
+      var updateBgLayerStyleUrlParam = function(bgLayer) {
         var styleUrlValue = bgLayer.externalStyleUrl;
         if (bgLayer.id) {
           getBgById(bgLayer.id).styleUrl = styleUrlValue;
```

Comment:
we should avoid having to deal with `externalStyleUrl` + `styleUrl` at the moment it's a **HUGE** mess in the background directive.
where is the source of truth? We don't know anymore, the layer in the map? the properties in the dict `vtBgs` that we extend on the fly?
It's getting really ugly...

Reply:
In the directive I am relying on the styleUrl which was used the first approach.
Try to change the background with a custom style and everything goes to hell.

Reply:
I gave it some thought, i think we could handle the style in gaGLStyle on a per layer basis, otherwise we won't be able to store mutiple map states. I propose that we also store the source url of the style if it differs from the original one (or even all the time). The problem with externalStyleUrl is that it is not persistent, each time we switch the background we loose this information.
What do you think?